lib/contracts.js

```diff
@@ -0,0 +1,83 @@
+import { requireContract } from "./utils.js";
+
+/**
+ * These are uninitialized instances of ExtendTruffleContract,
+ * effectively class factories.
+ */
+const GenesisScheme = requireContract("GenesisScheme");
+import { GlobalConstraintRegistrar } from   './globalconstraintregistrar.js';
+import { SchemeRegistrar } from   './schemeregistrar.js';
+import { ContributionReward } from   './contributionreward.js';
+import { AbsoluteVote} from "./absoluteVote.js";
+import { TokenCapGC} from "./tokenCapGC.js";
+import { UpgradeScheme } from   './upgradescheme.js';
+
+export async function getDeployedContracts() {
+  /**
+   * These are deployed contract instances represented by their respective Arc
+   * javascript wrappers (ExtendTruffleContract).
+   *
+   * `deployed()` is a static method on each of those classes.
+   **/
+  const contributionReward = await ContributionReward.deployed();
+  const genesisScheme = await GenesisScheme.deployed();
+  const globalConstraintRegistrar = await GlobalConstraintRegistrar.deployed();
+  const schemeRegistrar = await SchemeRegistrar.deployed();
+  const tokenCapGC = await TokenCapGC.deployed();
+  const upgradeScheme = await UpgradeScheme.deployed();
+  const absoluteVote = await AbsoluteVote.deployed();
+
+  /**
+   * `contract` here is an uninitialized instance of ExtendTruffleContract,
+   * basically the class factory.
+   * Calling contract.at() (a static method on the class) will return a
+   * the properly initialized instance of ExtendTruffleContract.
+   */
+  const contracts = {
```

Comment:
I'm really not a big fan of this dictionary style of storing the contracts, as an object whose property names are the names of contracts.  But at this moment I am at a loss for a better idea.  Maybe it is best.

Reply:
me either, this is just a start

Reply:
With a fresh morning mind it occurred to me that this contracts object should perhaps share at least some of the structure of the data that we are going to provide about current and historical contracts/schemes.  They two sets of data should be compatible.

Reply:
With a fresh morning mind it occurred to me that this contracts object should perhaps share at least some of the structure of the data that we are going to provide about current and historical contracts/schemes. They two sets of data should be compatible in schema.

Reply:
totally agree

Reply:
Is the plan then to refactor this when we do the work on supplying the full historical contracts?

Reply:
yeah